src/main/preload/externalAPI.ts

```diff
@@ -0,0 +1,470 @@
+        const extraParam = (new URLSearchParams(window.location.search)).get('extra');
+        isExpired = extraParam === 'expired';
+    } else {
+        isExpired = false;
+    }
+    if (isExpired !== sessionExpired) {
+        sessionExpired = isExpired;
+        ipcRenderer.send(SESSION_EXPIRED, sessionExpired);
+    }
+};
+
+/**
+ * Legacy message passing code - can be running alongside the new API stuff
+ */
+
+// Disabling no-explicit-any for this legacy code
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+window.addEventListener('message', ({origin, data = {}}: {origin?: string; data?: {type?: string; message?: any}} = {}) => {
+    const {type, message = {}} = data;
+    if (origin !== window.location.origin) {
+        return;
+    }
+    switch (type) {
+    case 'webapp-ready':
+    case 'get-app-version': {
+        // register with the webapp to enable custom integration functionality
+        ipcRenderer.invoke(GET_APP_INFO).then((info) => {
+            console.log(`registering ${info.name} v${info.version} with the server`);
+            window.postMessage(
+                {
+                    type: 'register-desktop',
+                    message: info,
+                },
+                window.location.origin || '*',
+            );
+        });
+        break;
+    }
+    case 'dispatch-notification': {
+        const {title, body, channel, teamId, url, silent, data: messageData} = message;
+        channels.set(channel.id, channel);
+        ipcRenderer.send(NOTIFY_MENTION, title, body, channel.id, teamId, url, silent, messageData.soundName);
+        break;
+    }
+    case BROWSER_HISTORY_PUSH: {
+        const {path} = message as {path: string};
+        ipcRenderer.send(BROWSER_HISTORY_PUSH, path);
+        break;
+    }
+    case 'history-button': {
+        ipcRenderer.invoke(REQUEST_BROWSER_HISTORY_STATUS).then(sendHistoryButtonReturn);
+        break;
+    }
+    case CALLS_LINK_CLICK: {
+        ipcRenderer.send(CALLS_LINK_CLICK, message.link);
+        break;
+    }
+    case GET_DESKTOP_SOURCES: {
+        ipcRenderer.invoke(GET_DESKTOP_SOURCES, message).then(sendDesktopSourcesResult);
+        break;
+    }
+    case CALLS_WIDGET_SHARE_SCREEN: {
+        ipcRenderer.send(CALLS_WIDGET_SHARE_SCREEN, message.sourceID, message.withAudio);
+        break;
+    }
+    case CALLS_JOIN_CALL: {
+        ipcRenderer.invoke(CALLS_JOIN_CALL, message).then(sendCallsJoinedCall);
+        break;
+    }
+    case CALLS_JOINED_CALL: {
+        ipcRenderer.send(CALLS_JOINED_CALL, message.callID);
+        break;
+    }
+    case CALLS_JOIN_REQUEST: {
+        ipcRenderer.send(CALLS_JOIN_REQUEST, message.callID);
+        break;
+    }
+    case CALLS_WIDGET_RESIZE: {
+        ipcRenderer.send(CALLS_WIDGET_RESIZE, message.width, message.height);
+        break;
+    }
+    case CALLS_ERROR: {
+        ipcRenderer.send(CALLS_ERROR, message.error, message.callID, message.errMsg);
+        break;
+    }
```

Comment:
Small typo. It's `message.err` what we need to pass through.

Reply:
Good catch :)